### PR TITLE
IArray extension methods

### PR DESCRIPTION
**Pull request by brunnerant**
This is work in progress, it doesn't work yet (in particular, the test time out for some unknown reason).

### CODE REVIEW

**Comment by brunnerant** [comment]
Should I copy all the documentation from `ArrayOps.scala` to the extension methods that I added ?

**Comment by nicolasstucki** [comment]
Yes, that would be great

**Comment by nicolasstucki** [comment]
BTW, #7806 will move the IArray file. This should in theory not be a conflict as it does not change its contents.

File: library/src-bootstrapped/scala/IArray.scala [hunk 1]
```diff
@@ -38,6 +38,166 @@ object opaques
     def (arr: IArray[Double]) length: Int = arr.asInstanceOf[Array[Double]].length
     def (arr: IArray[Object]) length: Int = arr.asInstanceOf[Array[Object]].length
     def [T](arr: IArray[T]) length: Int = arr.asInstanceOf[Array[T]].length
+
+    /** All the methods on Array[T] that don't mutate in-place can be used with IArray[T].
+     */
+    def [T, U >: T: ClassTag](arr: IArray[T]) ++(that: IArray[U]): IArray[U] =
+      (genericArrayOps(arr) ++ that.asInstanceOf[Array[U]]).asInstanceOf[IArray[U]]
```
Comment (nicolasstucki) [inline library/src-bootstrapped/scala/IArray.scala]: Are all the `asInstanceOf` still required?

Comment (brunnerant) [reply library/src-bootstrapped/scala/IArray.scala]: The `that.asInstanceOf[...]`, yes. The ones at the end are also required, but perhaps I can remove the brackets and it stills works.

**Comment by nicolasstucki** [comment]
The code LGTM. Just missing the documentation.

**Comment by brunnerant** [comment]
Ok, I will do it as soon as possible !

**Comment by brunnerant** [comment]
I just copied the comments from `ArrayOps.scala` (in Scala 2). Note that I only took the short description of the methods, not the entire documentation about the parameters and the return value.

**Pull request merged**